ext/date/lib/parse_date.c

```diff
@@ -171,7 +171,14 @@ static const timelib_tz_lookup_table timelib_timezone_utc[] = {
 };
 
 #if defined(_POSIX_TZNAME_MAX)
-# define MAX_ABBR_LEN _POSIX_TZNAME_MAX
+   /* Solaris exposes _POSIX_TZNAME_MAX = 3 unless _XPG6 is defined.
```

Comment:
you could not know, that needs to be a PR on its own [here](https://github.com/derickr/timelib).